### ISSUE TITLE
Calabash IOS fails with error "No connected devices" when UITextField is highlighted on screen

### ISSUE THREAD

**Issue by ravikunaparaju** [calabash/calabash-ios#751]
Issue Back ground:

The scenario is user navigates to redeem code screen, enter code and taps on redeem button.
When calabash navigates to redeem screen it fails with error "No Connected devices". 

Environment:

XCode 6.2
iPhone OS 8.2
Calabash-IOS 0.13.0

### Debug Logs:

```
INFO: Using uia strategy: 'host'
2015-04-23 10:17:39 +0100 [RunLoop:debug]: Preparation took 0.069397 seconds
2015-04-23 10:17:39 +0100 [RunLoop:debug]: 
{
                        :app => "/Users/rkunaparaju/code/os-maps-automation/apps/MyApp-cal.app",
                       :args => [],
    :bundle_dir_or_bundle_id => "my.bundle.id",
                  :bundle_id => "my.bundle.id",
                     :device => "iphone",
              :device_target => "ec9b055ed00a03f1e222a65781d19273f87d352a",
              :launch_method => :instruments,
             :launch_retries => 5,
                   :log_file => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187/run_loop.out",
                  :no_launch => false,
                    :no_stop => false,
                      :reset => false,
                :results_dir => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187",
          :results_dir_trace => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187/trace",
                     :script => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187/_run_loop.js",
                :sdk_version => nil,
                       :udid => "ec9b055ed00a03f1e222a65781d19273f87d352a",
               :uia_strategy => :host,
                      :xcode => "6.2",
                 :xcode_path => "/Applications/Xcode.app/Contents/Developer"
}

### Starting on ec9b055ed00a03f1e222a65781d19273f87d352a App: my.bundle.id ###
2015-04-23 10:17:40 +0100 [RunLoop:debug]: xcrun instruments -w ec9b055ed00a03f1e222a65781d19273f87d352a -D /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187/trace -t Automation my.bundle.id -e UIARESULTSPATH /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187 -e UIASCRIPT /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187/_run_loop.js >& /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150423-87135-1dfe187/run_loop.out
2015-04-23 10:17:43 +0100 [RunLoop:debug]: Launching took 3.852001 seconds
= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 94
Host: 192.168.2.3:37265

{"query":"OSRoundedButton marked:'Log in'","operation":{"method_name":"query","arguments":[]}}

= Response

! CONNECTION CLOSED
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 94
Host: 192.168.2.3:37265

{"query":"OSRoundedButton marked:'Log in'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 66
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[],"outcome":"SUCCESS"}= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 85
Host: 192.168.2.3:37265

{"query":"UIButton marked:'MENU'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 458
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UIButton: 0x13ed214d0; frame = (18 23; 39 44); opaque = NO; autoresize = RM+BM; layer = <CALayer: 0x17023a1c0>>","label":"MENU","frame":{"y":23,"x":18,"width":39,"height":44},"accessibilityElement":true,"value":null,"alpha":1,"enabled":true,"visible":1,"selected":false,"class":"UIButton","rect":{"y":23,"height":44,"center_y":45,"x":18,"center_x":37.5,"width":39}}],"outcome":"SUCCESS"}= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 85
Host: 192.168.2.3:37265

{"query":"UIButton marked:'MENU'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Thu, 23 Apr 2015 09:17:43 GMT
Content-Length: 458
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UIButton: 0x13ed214d0; frame = (18 23; 39 44); opaque = NO; autoresize = RM+BM; layer = <CALayer: 0x17023a1c0>>","label":"MENU","frame":{"y":23,"x":18,"width":39,"height":44},"accessibilityElement":true,"value":null,"alpha":1,"enabled":true,"visible":1,"selected":false,"class":"UIButton","rect":{"y":23,"height":44,"center_y":45,"x":18,"center_x":37.5,"width":39}}],"outcome":"SUCCESS"}Sending UIA command
uia.tapOffset('{:x 37.5, :y 45}')
2015-04-23 10:17:43 +0100 [RunLoop:debug]: 1:uia.tapOffset('{:x 37.5, :y 45}')
2015-04-23 10:17:43 +0100 [RunLoop:debug]: Trying write of command 1:uia.tapOffset('{:x 37.5, :y 45}') at index 1
2015-04-23 10:17:45 +0100 [RunLoop:debug]: validate index written for index 1 ok
Result
{"status"=>"success", "value"=>true, "index"=>1}
= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Thu, 23 Apr 2015 09:17:46 GMT
Content-Length: 102
Host: 192.168.2.3:37265

{"query":"UITableViewLabel  marked:'Redeem a Map'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Thu, 23 Apr 2015 09:17:46 GMT
Content-Length: 524
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UITableViewLabel: 0x13ed3a610; frame = (30 0; 374 44); text = 'Redeem a Map'; userInteractionEnabled = NO; layer = <_UILabelLayer: 0x1700931a0>>","label":"Redeem a Map","frame":{"y":0,"x":30,"width":374,"height":44},"accessibilityElement":true,"value":"Redeem a Map","alpha":1,"enabled":true,"visible":1,"text":"Redeem a Map","class":"UITableViewLabel","rect":{"y":458,"height":44,"center_y":480,"x":30,"center_x":217,"width":374}}],"outcome":"SUCCESS"}Sending UIA command
uia.tapOffset('{:x 217, :y 480}')
2015-04-23 10:17:46 +0100 [RunLoop:debug]: 2:uia.tapOffset('{:x 217, :y 480}')
2015-04-23 10:17:46 +0100 [RunLoop:debug]: Trying write of command 2:uia.tapOffset('{:x 217, :y 480}') at index 2
2015-04-23 10:17:47 +0100 [RunLoop:debug]: validate index written for index 2 ok
Result
{"status"=>"success", "value"=>true, "index"=>2}
    Given user is in Redeem Map Screen                                                              # features/step_definitions/redeem_steps.rb:60
    And user selects taps cancel button                                                             # features/step_definitions/redeem_steps.rb:65
      No connected devices (RuntimeError)
      ./features/IOS/Redeem.rb:48:in `cancelCode'
      ./features/step_definitions/redeem_steps.rb:66:in `/^user selects taps cancel button$/'
      features/redeemMap.feature:43:in `And user selects taps cancel button'
    Then app displays My Routes screen                                                              # features/step_definitions/redeem_steps.rb:56
Sending 'QUIT' to instruments process '87781'
Waiting for instruments '87781' to terminate
Waited for 0.933273 seconds for instruments with '87781' to terminate
 ```

**Comment by ravikunaparaju** [comment]
Using calabash-ios console, I am able to navigate to this screen, enter text in UITextField. All works fine

enter_text("UITextField index:0","AHS1")

It fails only when I run  'cucumber features/redeem.feature'

**Comment by jmoody** [comment]
We'll need more information.

```
  And user selects taps cancel button                                                             # features/step_definitions/redeem_steps.rb:65
      No connected devices (RuntimeError)
      ./features/IOS/Redeem.rb:48:in `cancelCode'
```

I don't recognize the `No connected devices` error message.

Can you paste the `redeem_steps.rb` Step?

**Comment by ravikunaparaju** [comment]
Hi Joshua,

Thanks for your response, The error message is inconsistent from device and simulator. 

When I reported the issue I am getting the error message "No device connected", now I am getting error 
"HTTPClient::KeepAliveDisconnected:  (HTTPClient::KeepAliveDisconnected)" on simulator and 
On Device the error is " A JSON text must at least contain two octets! (JSON::ParserError)"

I am not sure if my issue is related to https://github.com/calabash/calabash-ios/issues/259
Workarounds suggested there didn't worked for me. I still get same failures.

I have reproduce the error for you again on device and attaching  my feature, steps file and custom code as jpg files. 

As you see in the log it shows connection closed when calabash try to enter text in edit text fields. I can use the same commands to enter text in edit text field from console

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //map HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 88
Host: 192.168.2.3:37265

{"query":"UITextField index:0","operation":{"method_name":"query","arguments":["JNPQ"]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 73
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":["*****"],"outcome":"SUCCESS"}! CONNECTION CLOSED
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //map HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 88
Host: 192.168.2.3:37265

{"query":"UITextField index:1","operation":{"method_name":"query","arguments":["LHBO"]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 73
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":["*****"],"outcome":"SUCCESS"}! CONNECTION CLOSED
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //map HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 88
Host: 192.168.2.3:37265

{"query":"UITextField index:2","operation":{"method_name":"query","arguments":["XJJN"]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 73
Access-Control-Allow-Origin: *
![redeemfeature](https://cloud.githubusercontent.com/assets/11389188/7349427/d8e83fc2-ecf0-11e4-919d-c924f42715a3.jpg)
![redeem_steps](https://cloud.githubusercontent.com/assets/11389188/7349430/dbf613d8-ecf0-11e4-81b1-fe136543b722.jpg)
![redeemioscustomcode](https://cloud.githubusercontent.com/assets/11389188/7349431/de55d122-ecf0-11e4-9ef2-da8e98084427.jpg)

**Comment by ravikunaparaju** [comment]
Below is complete log for my feature run on device

DEVICE_ENDPOINT=http://192.168.2.3:37265 DEVICE_TARGET='ec9b055ed00a03f1e222a65781d19273f87d352a' APP_BUNDLE_PATH=my.app.bundle DEBUG=1 DEBUG_HTTP=1 cucumber -p ios features/redeemMap.feature
ravi:my-maps-automation rkunaparaju$ DEVICE_ENDPOINT=http://192.168.2.3:37265 DEVICE_TARGET='ec9b055ed00a03f1e222a65781d19273f87d352a' DEBUG=1 DEBUG_HTTP=1  cucumber -p ios features/redeemMap.feature --tags @wip
Using the ios profile...
Feature: A registered user should be successfuly redeem a map

  @wip
  Scenario: User enters a valid smart map code And redeem map # features/redeemMap.feature:17

INFO: Using uia strategy: 'host'
Sending 'QUIT' to instruments process '88181'
Waiting for instruments '88181' to terminate
Waited for 2.059989 seconds for instruments with '88181' to terminate
Sending 'KILL' to instruments process '88181'
Waiting for instruments '88181' to terminate
Waited for 0.10299 seconds for instruments with '88181' to terminate
2015-04-27 14:55:10 +0100 [RunLoop:debug]: Preparation took 2.236563 seconds
2015-04-27 14:55:10 +0100 [RunLoop:debug]: 
{
                        :app => "/Users/rkunaparaju/code/app-automation/apps/myApp-cal.app",
                       :args => [],
    :bundle_dir_or_bundle_id => "my.app.bundle",
                  :bundle_id => "my.app.bundle",
                     :device => "iphone",
              :device_target => "ec9b055ed00a03f1e222a65781d19273f87d352a",
              :launch_method => :instruments,
             :launch_retries => 5,
                   :log_file => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh/run_loop.out",
                  :no_launch => false,
                    :no_stop => false,
                      :reset => false,
                :results_dir => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh",
          :results_dir_trace => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh/trace",
                     :script => "/var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh/_run_loop.js",
                :sdk_version => nil,
                       :udid => "ec9b055ed00a03f1e222a65781d19273f87d352a",
               :uia_strategy => :host,
                      :xcode => "6.2",
                 :xcode_path => "/Applications/Xcode.app/Contents/Developer"
}

### Starting on ec9b055ed00a03f1e222a65781d19273f87d352a App: my.app.bundle ###
2015-04-27 14:55:11 +0100 [RunLoop:debug]: xcrun instruments -w ec9b055ed00a03f1e222a65781d19273f87d352a -D /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh/trace -t Automation my.app.bundle -e UIARESULTSPATH /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh -e UIASCRIPT /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh/_run_loop.js >& /var/folders/r1/dk681gq93z70dvdftkqk044w0000gp/T/run_loop20150427-88281-y537bh/run_loop.out
2015-04-27 14:55:15 +0100 [RunLoop:debug]: Launching took 4.251926 seconds
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:15 GMT
Content-Length: 94
Host: 192.168.2.3:37265

{"query":"OSRoundedButton marked:'Log in'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:15 GMT
Content-Length: 66
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[],"outcome":"SUCCESS"}= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:15 GMT
Content-Length: 85
Host: 192.168.2.3:37265

{"query":"UIButton marked:'MENU'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:15 GMT
Content-Length: 458
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UIButton: 0x154621e50; frame = (18 23; 39 44); opaque = NO; autoresize = RM+BM; layer = <CALayer: 0x170221f20>>","label":"MENU","frame":{"y":23,"x":18,"width":39,"height":44},"accessibilityElement":true,"value":null,"alpha":1,"enabled":true,"visible":1,"selected":false,"class":"UIButton","rect":{"y":23,"height":44,"center_y":45,"x":18,"center_x":37.5,"width":39}}],"outcome":"SUCCESS"}= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:15 GMT
Content-Length: 85
Host: 192.168.2.3:37265

{"query":"UIButton marked:'MENU'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:15 GMT
Content-Length: 458
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UIButton: 0x154621e50; frame = (18 23; 39 44); opaque = NO; autoresize = RM+BM; layer = <CALayer: 0x170221f20>>","label":"MENU","frame":{"y":23,"x":18,"width":39,"height":44},"accessibilityElement":true,"value":null,"alpha":1,"enabled":true,"visible":1,"selected":false,"class":"UIButton","rect":{"y":23,"height":44,"center_y":45,"x":18,"center_x":37.5,"width":39}}],"outcome":"SUCCESS"}Sending UIA command
uia.tapOffset('{:x 37.5, :y 45}')
2015-04-27 14:55:15 +0100 [RunLoop:debug]: 1:uia.tapOffset('{:x 37.5, :y 45}')
2015-04-27 14:55:15 +0100 [RunLoop:debug]: Trying write of command 1:uia.tapOffset('{:x 37.5, :y 45}') at index 1
2015-04-27 14:55:16 +0100 [RunLoop:debug]: validate index written for index 1 ok
Result
{"status"=>"success", "value"=>true, "index"=>1}
= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:17 GMT
Content-Length: 102
Host: 192.168.2.3:37265

{"query":"UITableViewLabel  marked:'Redeem a Map'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:17 GMT
Content-Length: 524
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UITableViewLabel: 0x154542a90; frame = (30 0; 374 44); text = 'Redeem a Map'; userInteractionEnabled = NO; layer = <_UILabelLayer: 0x1740990a0>>","label":"Redeem a Map","frame":{"y":0,"x":30,"width":374,"height":44},"accessibilityElement":true,"value":"Redeem a Map","alpha":1,"enabled":true,"visible":1,"text":"Redeem a Map","class":"UITableViewLabel","rect":{"y":502,"height":44,"center_y":524,"x":30,"center_x":217,"width":374}}],"outcome":"SUCCESS"}= Request

POST /map HTTP/1.1
Content-Type: application/x-www-form-urlencoded
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:17 GMT
Content-Length: 102
Host: 192.168.2.3:37265

{"query":"UITableViewLabel  marked:'Redeem a Map'","operation":{"method_name":"query","arguments":[]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:17 GMT
Content-Length: 524
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":[{"id":null,"description":"<UITableViewLabel: 0x154542a90; frame = (30 0; 374 44); text = 'Redeem a Map'; userInteractionEnabled = NO; layer = <_UILabelLayer: 0x1740990a0>>","label":"Redeem a Map","frame":{"y":0,"x":30,"width":374,"height":44},"accessibilityElement":true,"value":"Redeem a Map","alpha":1,"enabled":true,"visible":1,"text":"Redeem a Map","class":"UITableViewLabel","rect":{"y":502,"height":44,"center_y":524,"x":30,"center_x":217,"width":374}}],"outcome":"SUCCESS"}Sending UIA command
uia.tapOffset('{:x 217, :y 524}')
2015-04-27 14:55:17 +0100 [RunLoop:debug]: 2:uia.tapOffset('{:x 217, :y 524}')
2015-04-27 14:55:17 +0100 [RunLoop:debug]: Trying write of command 2:uia.tapOffset('{:x 217, :y 524}') at index 2
2015-04-27 14:55:18 +0100 [RunLoop:debug]: validate index written for index 2 ok
Result
{"status"=>"success", "value"=>true, "index"=>2}
    Given user is in  Redeem Map Screen                       # features/step_definitions/redeem_steps.rb:27
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //map HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 88
Host: 192.168.2.3:37265

{"query":"UITextField index:0","operation":{"method_name":"query","arguments":["JNPQ"]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 73
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":["*****"],"outcome":"SUCCESS"}! CONNECTION CLOSED
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //map HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 88
Host: 192.168.2.3:37265

{"query":"UITextField index:1","operation":{"method_name":"query","arguments":["LHBO"]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 73
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":["*****"],"outcome":"SUCCESS"}! CONNECTION CLOSED
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //map HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 88
Host: 192.168.2.3:37265

{"query":"UITextField index:2","operation":{"method_name":"query","arguments":["XJJN"]}}

= Response

HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Max-Age: 3000
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 73
Access-Control-Allow-Origin: *

{"status_bar_orientation":"down","results":["*****"],"outcome":"SUCCESS"}    When user has entered a valid smartmap redemption code    # features/step_definitions/redeem_steps.rb:32
! CONNECTION CLOSED
= Request

! CONNECT TO 192.168.2.3:37265
! CONNECTION ESTABLISHED
POST //gesture HTTP/1.1
Content-Type: application/json;charset=utf-8
User-Agent: HTTPClient/1.0 (2.6.0.1, ruby 2.0.0 (2014-05-08))
Accept: */*
Date: Mon, 27 Apr 2015 13:55:19 GMT
Content-Length: 186
Host: 192.168.2.3:37265

{"query_timeout":5.0,"gestures":[{"query_string":"* marked:'Code Entry'","touches":[{"x":50,"y":50,"offset_x":0,"offset_y":0,"wait":0.0,"time":0.2,"release":true,"query_string":null}]}]}

= Response

HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Connection: close
Accept-Ranges: bytes
Date: Mon, 27 Apr 2015 13:55:19 GMT

! CONNECTION CLOSED
    And user has selected Redeem                              # features/step_definitions/redeem_steps.rb:36
      A JSON text must at least contain two octets! (JSON::ParserError)
      ./features/IOS/Redeem.rb:42:in `submitCode'
      ./features/step_definitions/redeem_steps.rb:37:in `/^user has selected Redeem$/'
      features/redeemMap.feature:21:in `And user has selected Redeem'
    Then app displays Success screen with success message     # features/step_definitions/redeem_steps.rb:40
Sending 'QUIT' to instruments process '88304'
Waiting for instruments '88304' to terminate
Waited for 0.721828 seconds for instruments with '88304' to terminate

Failing Scenarios:
cucumber -p ios features/redeemMap.feature:17 # Scenario: User enters a valid smart map code And redeem map

1 scenario (1 failed)
4 steps (1 failed, 1 skipped, 2 passed)
0m12.232s

**Comment by jmoody** [comment]
`enter_text` is a relatively new feature in Calabash.

Can you try to use this pattern instead on iOS:

```
tap("UITextField index:0")
wait_for_keyboard
keyboard_enter_text(< your text >)
```

This looks wrong:

```
{"query":"UITextField index:0","operation":{"method_name":"query","arguments":["JNPQ"]}}
```

which makes me think that `enter_text` is buggy.

In the future, please wrap logs with ``` to make them more readable.

I am more interested in the exact Calabash call rather than the HTTP traffic.

**Comment by ravikunaparaju** [comment]
@jmoody,  It didn't worked with "keyboard_enter_text(< your text >)" either, when I observed this issue.  Strange thing is it works fine while debugging in console but fails when running the tests.

On the whole application calabash failed in two scenarios for which we flagged them as manual tests and moved on. 

Unfortunately I am not working on product any more.

Thanks to your team for this wonderful tool, great job on support :+1:
